src/common/transformations/src/transformations/common_optimizations/remove_multi_subgraph_op_dangling_params.cpp

```diff
@@ -157,6 +159,14 @@ bool ov::pass::RemoveMultiSubGraphOpDanglingParamsResults::run_on_model(const st
                         }
                     }
                 }
+                for (auto input_index : required_inputs_indices) {
+                    if (input_index > removed_loop_idx) {
+                        new_required_inputs_indices.insert(input_index - 1);
+                    } else {
+                        new_required_inputs_indices.insert(input_index);
+                    }
+                }
+                required_inputs_indices = new_required_inputs_indices;
```

Comment:
I thinks it's better to erase/insert then a creation a new set

Reply:
I am afraid that we iterator through container and erasing in the fly (during iteration) is not good